flake.nix

```diff
@@ -0,0 +1,77 @@
+{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+    fenix.url = "github:nix-community/fenix";
+    fenix.inputs.nixpkgs.follows = "nixpkgs";
+    devenv.url = "github:cachix/devenv";
+    flake-utils.url = "github:numtide/flake-utils";
+    crane = {
+      url = "github:ipetkov/crane";
+      inputs = {
+        flake-utils.follows = "flake-utils";
+        nixpkgs.follows = "nixpkgs";
+      };
+    };
+  };
+
+  outputs = { self, nixpkgs, devenv, fenix, flake-utils, crane, ... } @ inputs:
+    flake-utils.lib.eachSystem [ flake-utils.lib.system.x86_64-linux ]
+      (system:
+        let
+          pkgs = import nixpkgs { inherit system; };
+          toolchain = (fenix.packages.${system}.fromToolchainFile {
+            file = ./rust-toolchain.toml;
+            sha256 = "sha256-gdYqng0y9iHYzYPAdkC/ka3DRny3La/S5G8ASj0Ayyc=";
+          });
+          craneLib = crane.lib.${system}.overrideToolchain toolchain;
+
+          src = craneLib.cleanCargoSource (craneLib.path ./.);
+
+          env = {
+            OPENSSL_LIB_DIR = "${pkgs.lib.getLib pkgs.openssl}/lib";
+            OPENSSL_INCLUDE_DIR = "${pkgs.openssl.dev}/include";
+          };
+
+          buildInputs = with pkgs; [
+            postgresql
+          ];
+
+          commonArgs = {
+            inherit buildInputs src env;
+          };
+
+          cargoArtifacts = craneLib.buildDepsOnly commonArgs;
+
+          lemmy = craneLib.buildPackage (commonArgs // {
+            inherit cargoArtifacts;
+          });
+        in
+        {
+          # TODO: nix build can't build the flake that has a submodule dependency
```

Comment:
Is there an open issue for supporting submodules? May be a good idea to add a link here so it's easier to check on later.

Reply:
Done